components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -161,8 +170,20 @@ class Nimbus : NimbusApi {
 
     override fun updateExperiments() {
         if (!isInitialized) return
-        nimbus.updateExperiments()
-        onExperimentUpdated?.invoke(nimbus.getActiveExperiments())
+        scope.launch {
+            nimbus.updateExperiments()
+            // Get the experiments to record in telemetry
+            nimbus.getActiveExperiments().let {
+                if (it.any()) {
+                    recordExperimentTelemetry(it)
+                    // The current plan is to have the nimbus-sdk updateExperiments() function
+                    // return a diff of the experiments that have been received, at which point we
+                    // can emit the appropriate telemetry events.
+                }
+                onExperimentUpdated?.invoke(it)
+            }
+            notifyObservers { onExperimentsFetched() }
```

Comment:
I'm not sure `onExperimentsFetched()` is right here. We've not only fetched the experiment definitions from the server, we've updated our enrolments into them.

The [API definition document][1] suggests: `onUpdatesApplied()`, an argument of `updated: List<EnrolledExperiment>` with the diff of updated experiments would be suitable here.

[1]: https://docs.google.com/document/d/1kchihPal0__A4VOAiPJarNuZXns5KhOHHfIeIzT6zfU/edit#